modules/google-speech/src/backend/client.ts

```diff
@@ -0,0 +1,148 @@
+import { SpeechClient, protos } from '@google-cloud/speech'
+import { TextToSpeechClient } from '@google-cloud/text-to-speech'
+import axios from 'axios'
+import * as sdk from 'botpress/sdk'
+import * as mm from 'music-metadata'
+
+import { Config } from '../config'
+import { Audio } from './audio'
+import { SAMPLE_RATES } from './constants'
+import {
+  IRecognitionAudio,
+  IRecognitionConfig,
+  ISynthesizeSpeechRequest,
+  Container,
+  Codec,
+  AudioEncoding
+} from './typings'
+
+const debug = DEBUG('google-speech')
+const debugSpeechToText = debug.sub('speech-to-text')
+const debugTextToSpeech = debug.sub('text-to-speech')
+
+export class GoogleSpeechClient {
+  private logger: sdk.Logger
+  private speechClient: SpeechClient
+  private textToSpeechClient: TextToSpeechClient
+
+  constructor(private bp: typeof sdk, private botId: string, private readonly config: Config) {
+    this.logger = this.bp.logger.forBot(this.botId)
+  }
+
+  public async initialize() {
+    if (!this.config.clientEmail || !this.config.privateKey) {
+      return this.logger.error(
+        '[GoogleSpeech] The clientEmail and privateKey must be configured in order to use this module.'
+      )
+    }
+
+    this.speechClient = new SpeechClient({
+      credentials: {
+        client_email: this.config.clientEmail,
+        private_key: this.config.privateKey
+      }
+    })
+
+    this.textToSpeechClient = new TextToSpeechClient({
+      credentials: {
+        client_email: this.config.clientEmail,
+        private_key: this.config.privateKey
+      }
+    })
```

Comment:
Will those throw if credentials are invalid ? If yes, please wrap them in try/catch to display meaningful message. If they dont, please add some sort of validation to check if it's valid